### ISSUE TITLE
Remove dependency on 'solc' from build process

### ISSUE THREAD

**Issue by akolotov** [poanetwork/poa-bridge#77]
Since bridge contracts are in separate project just now and deployed separately, 'solc' is not needed as dependency for the bridge.

**Comment by yrashk** [comment]
Bridge relies on these contracts to generate calls, transactions and event filters, so I don't think it's reasonable to drop this dependency at this moment. What will happen after we're done transitioning integration tests is that we can replace contracts in this project with interfaces from actual contracts to keep things leaner inside. But we'll still need `solc`.

**Comment by rstormsf** [comment]
maybe you only need ABIs  ? 
those are just JSON files

**Comment by rstormsf** [comment]
https://gist.github.com/rstormsf/998ae8acbe467788bdf95778fbc68cf0

3 files

**Comment by akolotov** [comment]
Agree with Roman. I think you could use git submodules to link to repos: poa-bridge and poa-parity-bridge-contracts.

**Comment by yrashk** [comment]
Yes, ABI will do if we want to get rid of `solc` dependendy. They will just make a tad harder to synchronize between sub-projects (instead of copying files, we'd need to use solc to compile them to .abi every time we sync). If we're willing to compromise here, then sure. Ideally, interfaces should not change too often.

@akolotov poa-parity-bridge-contracts doesn't seem to contain ABIs as is.

**Comment by rstormsf** [comment]
Correct, those are artifacts, which I already provided in the gist above

**Comment by yrashk** [comment]
As per @rstormsf it might be better to store those in https://github.com/poanetwork/poa-chain-spec

**Comment by akolotov** [comment]
Created https://github.com/poanetwork/poa-bridge-contracts/issues/38 to upload ABIs to specification repo
